source/extensions/filters/network/zookeeper_proxy/zookeeper_decoder.cc

```diff
@@ -0,0 +1,344 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_decoder.h"
+
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+void DecoderImpl::decode(Buffer::Instance& data, uint64_t& offset) {
+  ENVOY_LOG(trace, "zookeeper_proxy: decoding {} bytes at offset {}", data.length(), offset);
+
+  // Check message length.
+  int32_t len = BufferHelper::peekInt32(data, offset);
+  checkLength(len, 8);
+
+  // "Special" requests.
```

Comment:
Mind elaborating in the comment what makes these special?

Reply:
Done.